tools/src/h5diff/ph5diff_main.c

```diff
@@ -165,34 +166,44 @@ ph5diff_worker(int nID)
             diffs.nfound  = diff(file1_id, args.name1, file2_id, args.name2, &(args.opts), &(args.argdata));
             diffs.not_cmp = args.opts.not_cmp;
 
-            /* If print buffer has something in it, request print token.*/
-            if (outBuffOffset > 0) {
+            if ((outBuffOffset == 0) && !overflow_file)
+                /* Nothing to print. Send diffs to manager */
+                MPI_Send(&diffs, sizeof(diffs), MPI_BYTE, 0, MPI_TAG_DONE, MPI_COMM_WORLD);
```

Comment:
Moves the "nothing to print" case up here for clarity and also fixes an issue where output in the overflow file wasn't getting printed when `outBuffOffset == 0`.